tvaultuiv2/src/views/private/service-accounts/components/ServiceAccountPermission/components/Users/index.js

```diff
@@ -0,0 +1,293 @@
+  const [response, setResponse] = useState({ status: 'loading' });
+  const isMobileScreen = useMediaQuery(small);
+
+  // on svc account meta data is available.
+  useEffect(() => {
+    if (accountMetaData && Object.keys(accountMetaData).length !== 0) {
+      if (Object.keys(accountMetaData?.response).length !== 0) {
+        setResponse({ status: 'success' });
+      }
+    } else {
+      setResponse({ status: '' });
+    }
+  }, [accountMetaData]);
+
+  // When add permission button is clicked.
+  useEffect(() => {
+    if (newPermission) {
+      setResponse({ status: 'add' });
+    }
+  }, [newPermission]);
+
+  /**
+   * @function onDeleteClick
+   * @description function to delete the user from the svc account users list.
+   * @param {username} string username of the user.
+   * @param {access} string permission of the user.
+   */
+  const onDeleteClick = (username, access) => {
+    setResponse({ status: 'loading' });
+    const payload = {
+      access,
+      svcAccName: accountDetail.name,
+      username,
+    };
+    apiService
+      .deleteUserPermission(payload)
+      .then(async (res) => {
+        if (res && res.data?.messages && res.data.messages[0]) {
+          updateToastMessage(1, res.data.messages[0]);
+          setResponse({ status: '' });
+          refresh();
+        }
+      })
+      .catch((err) => {
+        setResponse({ status: 'success' });
+        if (err.response?.data?.errors && err.response.data.errors[0]) {
+          updateToastMessage(-1, err.response.data.errors[0]);
+        }
+      });
+  };
+
+  /**
+   * @function onSaveClicked
+   * @description function to save the user to the svc account users list.
+   * @param {data} object payload to call api.
+   */
+  const onSaveClicked = (data) => {
+    setResponse({ status: 'loading' });
+    apiService
+      .addUserPermission(data)
+      .then((res) => {
+        if (res && res.data?.messages) {
+          updateToastMessage(1, res.data?.messages[0]);
+          setResponse({ status: '' });
+          refresh();
+        }
+      })
+      .catch((err) => {
+        if (err.response?.data?.errors && err.response.data.errors[0]) {
+          updateToastMessage(-1, err.response.data.errors[0]);
+        }
+        setResponse({ status: 'success' });
+      });
+  };
+
+  const checkAccess = (access) => {
+    let val = '';
+    if (access === 'write') {
+      val = 'reset';
+    } else {
+      val = access;
+    }
+    return val;
+  };
+
+  /**
+   * @function onSubmit
+   * @description function structure the payload when save/edit is clicked and call save api.
+   * @param {username} string user name of the user.
+   * @param {access} string permission given to the user.
+   */
+  const onSubmit = async (username, access) => {
+    const value = {
+      access: checkAccess(access),
+      svcAccName: `${accountDetail.name}`,
+      username: username.toLowerCase(),
+    };
+    onSaveClicked(value);
```

Comment:
```suggestion
    onSaveClicked(value);
```

try {
 await fnCall();
} catch(e) {
}